### ISSUE TITLE
Any chance to apply those changes?

### ISSUE THREAD

**Issue by Najafov007** [robusta-dev/krr#337]
Hello! Thank you for your app, it's amazing! However, we do have some sort of an issue. We have our own cluster that runs approximately 1k pods on it. The thing is besides the report itself, is there any chance to apply those changes? Like its impossible to do it manually, isn't it? For example we use kustomize on our services and GitOps philosophy. How can we for example make it automatically apply those changes on a new branch? Would be amazing. Thank you

**Comment by aantn** [comment]
Hi, we don't have the option available but of course PRs are welcome! You could create a custom outputer that generates kustomize overlays, for example.

**Comment by wrbbz** [comment]
That's a nice feature to have.
We also got a lot of pods. It's nice to have some analytics from the tool, but it's very unpleasant to tweak resources manually. 

I thought about automatic PRs with resource changes. Some of which could be automerged. But for that thing, you have to know in which file to change resource. For that you have to have some kind of configuration file, which you need to parse. And implement a ton of logic for interaction with Git(Hub|Lab) API. and that logic will be out of scope of this tool

Another option - to apply suggestions on the fly. And somehow monitor diff between source code and actual resource. However, I suppose, you do not want to apply all suggestions (at least, I do not). It would be perfect to select which suggestions to apply. And thus, you also will need some kind of configuration

Both approaches require a ton of time and effort as I think

**Comment by aantn** [comment]
Yep! @Avi-Robusta from the Robusta team actually has a working PR that uses LLMs to figure out which file to change, and use the GitHub APIs to open a PR. (It is indeed a little out of scope for KRR - he's doing it with our other OSS project https://github.com/robusta-dev/holmesgpt/)

Let me know if you're interested in trying it out and testing.
